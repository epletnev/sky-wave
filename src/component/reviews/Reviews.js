import React, { useState } from "react";
import { FaPencilAlt } from "react-icons/fa";
import FeedBackModal from "../feedBackModal/FeedBackModal";
import "./Reviews.scss";

function Reviews({ reviews }) {
  const [isOpen, setIsOpen] = useState(false);
  return (
    <div className="reviews-container">
      <section className="review-section">
        <p>Reviews</p>
        <hr className="redLine" />
        <hr className="thickGreyLine" />
        <div className="reviews-title">
          <p>Leave A Review</p>
          <button onClick={() => setIsOpen(true)} className="pencil-icon">
            <FaPencilAlt />
          </button>
        </div>
        <hr className="thinGreyLine" />
        {reviews &&
          reviews.map((user) => (
            <div className="reviews" key={user.name}>
              <img
                src="/assets/images/avatars/avatar-1.png"
                alt="avatar"
                width="50"
                height="50"
              />
              <ul>
                <li className="name">{user.name}</li>
                <li className="date">
                  {new Date(user.date).toLocaleDateString("en-US", {
                    year: "numeric",
                    month: "short",
                    day: "numeric",
                    hour: "numeric",
                    minute: "numeric",
                  })}
                </li>
                <li className="message">{user.message}</li>
              </ul>
            </div>
          ))}
      </section>
      <FeedBackModal open={isOpen} onClose={() => setIsOpen(false)} />
    </div>
  );
}

export default Reviews;
