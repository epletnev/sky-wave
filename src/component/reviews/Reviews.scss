@import "../../theme/variables.scss";

.reviews-container {
  display: flex;
  justify-content: space-between;
}

.review-section {
  width: 60%;
  padding: 2rem;
  p {
    padding-left: 2.5rem;
  }
}

.reviews {
  display: flex;
  ul {
    list-style-type: none;
    padding: 1rem;
  }
  img {
    margin: 1rem;
  }
  .name {
    font-weight: bold;
  }
  .message {
    font-size: 0.8rem;
  }
  .date {
    color: gray;
    font-size: 0.8rem;
  }
}

.reviews-title {
  display: flex;
  justify-content: flex-end;
  p {
    padding-right: 14rem;
    margin: 1rem;
  }
  button {
    margin: 1rem;
  }
}

.pencil-icon {
  display: block;
  height: 30px;
  width: 30px;
  color: $theme-Color;
  background-color: pink;
  border-radius: 50%;
  border: none;
  padding-bottom: 5px;
  margin: 5px;
}

.redLine {
  border-top: 3px solid $theme-Color;
  margin: 0;
  margin-right: 80%;
}

.thickGreyLine {
  border-top: 2px solid lightgrey;
  margin: auto;
}

.thinGreyLine {
  border-top: 1px solid lightgrey;
  margin: auto;
}
