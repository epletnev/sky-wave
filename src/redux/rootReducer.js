import { combineReducers } from "redux";
import menuReducer from "./menu/menuReducer";
import menuDetailsReducer from "./menuDetails/menuDetailsReducer";
import aboutReducer from "./about/aboutReducer";
import contactReducer from "./contact/contactReducer";

const rootReducer = combineReducers({
  menu: menuReducer,
  menuDetails: menuDetailsReducer,
  about: aboutReducer,
  contact: contactReducer,
});

export default rootReducer;
