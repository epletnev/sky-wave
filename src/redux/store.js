import { createStore, applyMiddleware } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import rootReducer from "./rootReducer";
import Thunk from "redux-thunk";

export const getStore = () => {
  const store = createStore(
    rootReducer,
    composeWithDevTools(applyMiddleware(Thunk))
  );
  return store;
};
