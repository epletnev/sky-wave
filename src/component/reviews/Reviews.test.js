// import React from "react";
// import FeedBackModal from "./FeedBackModal";
// import renderer from "react-test-renderer";

// test("should render FeedBackModal component", () => {
//   const tree = renderer.create(<FeedBackModal />).toJSON();
//   expect(tree).toMatchSnapshot;
// });

import { create } from "react-test-renderer";
import { Provider } from "react-redux";
import { render, screen } from "@testing-library/react";
import { MemoryRouter as Router, Route } from "react-router-dom";
import { applyMiddleware, createStore } from "redux";
import thunk from "redux-thunk";
import rootReducer from "../../redux/rootReducer";
// import FeedBackModal from "../feedBackModal/FeedBackModal";
// import MenuDetailsCard from "./MenuDetailsCard";
// import Reviews from "../reviews/Reviews";
import Reviews from "./Reviews";

const mockedState = {
  menuDetails: {
    menuDetails: [],
    reviews: [],
  },
};

const mockStore = createStore(rootReducer, mockedState, applyMiddleware(thunk));

const MockedForm = () => (
  <Provider store={mockStore}>
    <Router>
      <Route path={`/menu/1`} component={Reviews} />
    </Router>
  </Provider>
);

describe("Our First Snapshot Test Case", () => {
  test("Testing the Tabs component UI for the App", () => {
    let main = create(MockedForm());
    expect(main.toJSON()).toMatchSnapshot();
  });
});
