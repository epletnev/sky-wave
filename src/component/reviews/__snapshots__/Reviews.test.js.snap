// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Our First Snapshot Test Case Testing the Tabs component UI for the App 1`] = `null`;
